internal/blobmanager/azureblob/backend.go

```diff
@@ -84,7 +84,8 @@ func resourceName(digest string) string {
 // Exists check that the artifact is already present in the repository
 func (b *Backend) Exists(ctx context.Context, digest string) (bool, error) {
 	_, err := b.Describe(ctx, digest)
-	if err != nil && errors.As(err, &backend.ErrNotFound{}) {
+	notFoundErr := &backend.ErrNotFound{}
+	if err != nil && errors.As(err, &notFoundErr) {
```

Comment:
can you please provide more info why this change here?

Reply:
The context is that this wasn't working before, as the error wasn't cast properly. I noticed it because of the tests in AWS which hit a similar code.